# Løsningsforslag - prissamarbeid

Vi skal her se på caset med bryggeriene Graff Brygghus og Bryggeri 13, og se om det vil lønne seg for disse å inngå et samarbeid.

Informasjon om markedet:

I markedet for mikroøl har vi to bryggerier i Tromsø, Graff Brygghus og Bryggeri 13. I produksjon av øl velger bedriftene produksjonsvolum i god tid før produktene er på markedet. 

Etterspørselen i markedet er gitt ved:

$$
P = 175 – 2Q
$$

hvor $P$ er markedspris for mikroøl per flaske, $Q$ er totalt kvantum (antall tusen flasker) og $Q = q_G + q_B$, der $q_G$ er produsert kvantum for Graff Brygghus og qB er produsert kvantum for Bryggeri 13. I produksjon av mikroøl vil Graff Brygghus og Bryggeri 13 ha konstante marginalkostnader på kr 10 perflaske, og begge bedriftene har årlige faste kostnader på kr 500 000.

Ved Cournot konkurranse i dette markedet vil optimal produksjon for hvert av bryggeriene være på 27 500 flasker, til en pris på kr 65 per flaske. Dette vil gi bedriftene en profitt på 1 012 500.

Anta at bryggeriene blir enige om å koordinere sin adferd. De avtaler å sette monopolpris og å dele markedet likt mellom seg. Videre avtaler de at hvis en av dem i en periode avviker fra denne avtalen, vil de begge i alle framtidige perioder opptre som konkurrenter.

a) Hva vil total produksjon av øl være hvis bryggeriene velger samarbeid og hvor mye vil bryggeriene tjene på dette?

In [112]:
import numpy as np
import sympy as sp

In [113]:
# Definerer symboler
a,b,q_g,q_b,c,Q,P,FK,r = sp.symbols('a b q_g q_b c Q P FK r', real=True, positive=True)

# Definerer invers etterspørsel
invers_etterspørsel = sp.Eq(P, 175-2*Q)

# Viser invers etterspørsel
invers_etterspørsel

Eq(P, 175 - 2*Q)

In [114]:
# Finner totalinntekt
TR = invers_etterspørsel.rhs * Q

# Viser totalinntekt
TR

Q*(175 - 2*Q)

In [115]:
# Deriverer totalinntekt for å finne marginalinntekt
MR = sp.diff(TR, Q)

# Viser marginalinntekt
MR

175 - 4*Q

In [116]:
# Definerer marginalkostnaden
c = 10

# Setter marginalinntekt lik marginalkostnad
likevekt = sp.Eq(MR, c)

# Viser likevekt
likevekt

Eq(175 - 4*Q, 10)

In [117]:
# Løser for optimalt kvantum for monopol
optimal_kvantum = sp.solve(likevekt, Q)[0]

# Viser optimalt kvantum for monopol
display(round(optimal_kvantum,2))

# Viser optimalt kvantum siden vi har to bedrifter som har prissamarbeid og da deler vi på to
display(round(optimal_kvantum,3)/2)

41.25

20.625

In [118]:
# Finner optimal pris ved å sette inn optimalt kvantum i invers etterspørsel
optimal_pris = invers_etterspørsel.subs(Q, optimal_kvantum).rhs

# Viser optimal pris, her deler vi ikke på to siden begge blir å sette samme pris
display(round(optimal_pris,2))

92.50

In [119]:
# Definerer faste kostnader, de er gitt i oppgaven og er på 500 000
FK = 500

# Finner profitt ved å bruke formelen for profitt
profitt = (invers_etterspørsel.subs(Q, optimal_kvantum).rhs - c) * optimal_kvantum

# Viser total profitt
display(round(profitt,1))

# Finner profitt per bedrift ved å dele på to og trekker fra faste kostnader
display(round(profitt/2-FK,1))

3403.1

1201.6

Merk at når vi regner i kode så har vi verdiene delt på tusener, så når vi skal forklare hva som blir total produksjon så må vi gange med 1000, det samme gjelder for profitt.

Det vi får her er at:

$$
Q^* = 41.25 \rightarrow 41.25 \times 1000 = 41250
$$

Så vi får derfor en total produksjon på 41250 flasker.

Prisen blir den samme for begge bryggeriene, og vi får derfor:

$$
P^* = 175 - 2 \times Q^* = 175 - 2 \times 41.25 = 92.5
$$

I oppgaven så spørres det om hvor mye bryggeriene tjener på å prissamarbeide, og tidligere under cournot konkurranse hadde de en profitt hver på 1 012 500.

For å finne profitten til bryggeriene så må vi bruke formelen for profitt:

$$
\pi = (P - MC) \times Q - FK \rightarrow \pi = (92.5-10) \times 41250 - 500000 = 1 201 600
$$

Derfor vil de tjene på prissamarbeid:

$$
\pi = 1 201 600 - 1 012 500 = 189 062.5
$$

Og fortjenesten per bryggeri blir da 189 062 kr.

b) Hvor mye vil Bryggeri 13 tjene på å avvike fra samarbeidsavtalen med Graff Brygghus i en periode?

In [120]:
Q = (q_g + q_b)

# Definerer invers etterspørsel
invers_etterspørsel = sp.Eq(P, 175-2*Q)

# Viser invers etterspørsel
invers_etterspørsel

Eq(P, -2*q_b - 2*q_g + 175)

In [121]:
# Finner totalinntekt
TR_1 = invers_etterspørsel.rhs * q_g
TR_2 = invers_etterspørsel.rhs * q_b

# Viser totalinntekt
display(TR_1, TR_2)

q_g*(-2*q_b - 2*q_g + 175)

q_b*(-2*q_b - 2*q_g + 175)

In [122]:
# Deriverer totalinntekt for å finne marginalinntekt
MR_1 = sp.diff(TR_1, q_g)
MR_2 = sp.diff(TR_2, q_b)

# Viser marginalinntekt
display(MR_1, MR_2)

-2*q_b - 4*q_g + 175

-4*q_b - 2*q_g + 175

In [123]:
# Setter marginalinntekt lik marginalkostnad
likevekt_qg = sp.Eq(MR_1, c)
likevekt_qb = sp.Eq(MR_2, c)

# Viser likevekt
display(likevekt_qg, likevekt_qb)

Eq(-2*q_b - 4*q_g + 175, 10)

Eq(-4*q_b - 2*q_g + 175, 10)

In [124]:
# Finner reaksjonsfunksjon for Graff Bryggeri
qb = sp.solve(likevekt_qg, q_g)[0]

display(qb)

165/4 - q_b/2

In [125]:
# Nå har vi funnet reaksjonsfunksjon til Graff bryggeri og da setter vi inn kvantumet
# Vi vet at hvis de samarbeider så ville optimalt kvantum være 41.25/2 = 20.625
ny_opt_q = qb.subs(q_b, round(optimal_kvantum,3)/2)

# Viser ny optimalt kvantum for Graff Bryggeri
ny_opt_q

30.938

In [126]:
# For å finne ny pris må vi sette inn ny kvantum i invers etterspørsel for Bryggeri 13, mens Graff beholder sitt kvantum
ny_opt_pris = invers_etterspørsel.subs({q_b:ny_opt_q,q_g:optimal_kvantum/2}).rhs

# Viser ny pris i markedet
ny_opt_pris

71.875

In [127]:
# Finner ny profitt for Bryggeri 13
ny_profitt = (ny_opt_pris - c) * ny_opt_q-FK

# Viser ny profitt for Graff Bryggeri
ny_profitt

1414.3

In [128]:
# Finner ny profitt for Graff Bryggeri
ny_profitt_graff = (ny_opt_pris - c) * (optimal_kvantum/2)-FK

# Viser ny profitt for Graff Bryggeri
ny_profitt_graff

776.17

Hvis Bryggeri 13 avviker fra samarbeidsavtalen, så vil de oppføre seg som Cournot konkurrenter.

Vi regner ved bruk av Cournot modellen, og finner til slutt reaksjonsfunksjonen til Graff bryggeri.

Orignalt ved prissamarbeid vet vi at begge bryggeriene ville produsert 20625 enheter:

$$
q_b = \frac{41.25}{2} =  20.625 \rightarrow 20.625 \times 1000 = 20625
$$

Derfor setter vi inn kvantumet i reaksjonsfunksjonen til Graff bryggeri, og da finner Bryggeri 13 ut hvor mye ny produksjon blir når de velger å avvike fra prissamarbeid:

$$
\frac{165}{4} - \frac{q_{b}}{2} \rightarrow \frac{165}{4} - \frac{20.625}{2} = 30.938 \times 1000 = 30938
$$

For å finne ny pris setter vi inn nye verdier for kvantum i invers etterspørsel:

$$
P = 175 (-2 \times (q_b + q_g)) \rightarrow P = 175 (-2 \times (20.625 + 30.938)) = 71.875
$$

Og den nye profitten for Bryggeri 13:

$$
\pi_b = (71.875-10) \times 30938 - 500000 =  1414288
$$

Mens Graff Bryggeri får nå:

$$
\pi_g = (71.875-10) \times 20625 - 500000 = 776171
$$

Merk at python kan noengang runde av tall slik at om du regner for hånd eller med kalkulator kan profitt bli mer presis enn det som kommer ut i python. Det er derfor viktig å være klar over at det kan være små forskjeller i svaret.

Om du endrer koden noe eller vil ha et mer presist svar kan du regne profitten med kalkulator, eller for hånd.

c) Hvor stor må diskonteringsfaktoren være for at bryggeriene skal ha incentiver til å opprettholde samarbeidsavtalen over tid?

In [129]:
from sympy import Symbol
from sympy.solvers.inequalities import reduce_rational_inequalities

x = Symbol('x', real=True)

In [130]:
reduce_rational_inequalities([[x + 2 > 0]], x)

-2 < x

In [131]:
# Nåverdi ved sammarbeid > nåverdi ved avvik - med p=1
reduce_rational_inequalities([[1201.6/(1-x) > 1414.258+(x*1012.5)/(1-x)]], x)

(0.529318644557171 < x) & (x < 1.0)

In [132]:
reduce_rational_inequalities([[1/(1+r) > 0.52931]], r)

r < 0.889252045115339

In [133]:
# Nåverdi ved sammarbeid > nåverdi ved avvik - med p=0.6
reduce_rational_inequalities([[1800/(1-(0.6*x)) > 2025+((0.6*x)*1600)/(1-(0.6*x))]], x)

(0.882352941176471 < x) & (x < 1.66666666666667)

In [134]:
reduce_rational_inequalities([[1/(1+r) > 0.88235]], r)

r < 0.133337111123704